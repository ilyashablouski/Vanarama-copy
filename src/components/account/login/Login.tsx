import React, { ChangeEvent } from 'react';
import { Row, Col } from 'react-grid-system';
import localForage from 'localforage';
import Input from '@vanarama/uibook/src/components/atoms/textinput';
import Button from '@vanarama/uibook/src/components/atoms/button';
import Link from '@vanarama/uibook/src/components/atoms/link';
import { LoginProps, LoginState } from './interfaces';

class Login extends React.Component<LoginProps, LoginState> {
  constructor(props: any) {
    super(props);

    this.state = {
      email: '',
      password: '',
      token: null,
    };
  }

  componentDidUpdate(_, prevState) {
    if (this.props.token) {
      localForage.setItem('va-token', this.props.token);
    }
  }

  handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const { email, password } = this.state;

    this.props.login(email, password);
  };

  handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
    e.preventDefault();
    const { name, value } = e.currentTarget;

    if (Object.keys(this.state).includes(name)) {
      this.setState(prevState => ({ ...prevState, [name]: value }));
    }
  };

  render() {
    const { email, password } = this.state;

    return (
      <section>
        <form onSubmit={this.handleSubmit} id="loginForm" className="form">
          <Row style={{ marginBottom: '16px' }}>
            <Col>
              <Input
                id="loginEmail"
                label="Your Email"
                type="text"
                name="email"
                value={email}
                handleChange={this.handleInputChange}
              />
            </Col>
          </Row>
          <Row style={{ marginBottom: '16px' }}>
            <Col>
              <Input
                id="loginPassword"
                label="Your Password"
                type="password"
                name="password"
                value={password}
                handleChange={this.handleInputChange}
              />
            </Col>
          </Row>
          <Row style={{ marginBottom: '16px' }}>
            <Col>
              <Link id="forgotPassword" href="password-request">
                forgot your password?
              </Link>
            </Col>
          </Row>
          <Row style={{ marginBottom: '16px' }}>
            <Col>
              <Button
                id="loginButton"
                type="submit"
                label="Login"
                color="primary"
              />
            </Col>
          </Row>
          <Row style={{ marginBottom: '16px' }}>
            <Col>
              {this.props.authenticated ? (
                <p id="loginStatus">Login Success</p>
              ) : null}
              {this.props.authenticated === false ? (
                <p id="loginFailure">Login Failed</p>
              ) : null}
            </Col>
          </Row>
        </form>
      </section>
    );
  }
}

export default Login;
