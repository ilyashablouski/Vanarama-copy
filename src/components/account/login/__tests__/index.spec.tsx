import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Login from '../index';
import {
  findForm,
  fillInputField,
} from '../../../../services/utils/test-helpers';

const email = 'email';
const password = 'password';

describe('<Login />', () => {
  const login = jest.fn();

  it('renders correctly with authentication', () => {
    const wrapper = mount(<Login login={login} authenticated token="" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without authentication', () => {
    const wrapper = mount(
      <Login login={login} authenticated={false} token="" />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('Submit', () => {
    let login;
    let wrapper;

    beforeEach(() => {
      login = jest.fn();
      wrapper = shallow(<Login login={login} authenticated={false} token="" />);
      fillLoginFields(wrapper);
      findForm(wrapper).simulate('submit', { preventDefault() {} });
    });

    afterEach(() => {
      jest.restoreAllMocks();
    });

    it('it is called once', () => {
      expect(login).toHaveBeenCalledTimes(1);
    });

    it('it is called with correct parameters', () => {
      expect(login).toBeCalledWith(email, password);
    });
  });

  describe('HandleChange', () => {
    let login;
    let wrapper;

    beforeEach(() => {
      login = jest.fn();
      wrapper = shallow(<Login login={login} authenticated={false} token="" />);
    });

    afterEach(() => {
      jest.restoreAllMocks();
    });

    it('it has correct state [email]', () => {
      fillLoginFields(wrapper);
      expect(wrapper.state(email)).toEqual(email);
    });

    it('it has correct state [password]', () => {
      fillLoginFields(wrapper);
      expect(wrapper.state(password)).toEqual(password);
    });

    it('it has incorrect state [email]', () => {
      expect(wrapper.state('email')).toEqual('');
    });

    it('it has incorrect state [password]', () => {
      expect(wrapper.state('password')).toEqual('');
    });
  });
});

function fillLoginFields(wrapper: any) {
  fillInputField(wrapper, 'loginEmail', { name: email, value: email });
  fillInputField(wrapper, 'loginPassword', { name: password, value: password });
}
