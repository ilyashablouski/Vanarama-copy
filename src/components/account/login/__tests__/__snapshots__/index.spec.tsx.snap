// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly with authentication 1`] = `
<Login
  authenticated={true}
  login={[MockFunction]}
  token=""
>
  <section>
    <form
      className="form"
      id="loginForm"
      onSubmit={[Function]}
    >
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="loginEmail"
                    label="Your Email"
                    name="email"
                    type="text"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="loginEmail-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="loginEmail"
                      >
                        <span
                          className="Text -primary"
                        >
                          Your Email
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="loginEmail"
                          name="email"
                          onChange={[Function]}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="loginPassword"
                    label="Your Password"
                    name="password"
                    type="password"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="loginPassword-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="loginPassword"
                      >
                        <span
                          className="Text -primary"
                        >
                          Your Password
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="loginPassword"
                          name="password"
                          onChange={[Function]}
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    color="primary"
                    id="forgotPassword"
                    plain={true}
                    size="regular"
                  >
                    <a
                      className="Link -primary -regular -plain"
                      id="forgotPassword"
                    >
                      forgot your password?
                    </a>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    color="primary"
                    fill="solid"
                    id="loginButton"
                    label="Login"
                    size="regular"
                    tabIndex={0}
                    type="submit"
                  >
                    <button
                      className="Button -primary -regular -solid"
                      id="loginButton"
                      tabIndex={0}
                      type="submit"
                    >
                      <div
                        className="Button__inner"
                      >
                        Login
                      </div>
                    </button>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <p
                    id="loginStatus"
                  >
                    Login Success
                  </p>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
    </form>
  </section>
</Login>
`;

exports[`<Login /> renders correctly without authentication 1`] = `
<Login
  authenticated={false}
  login={[MockFunction]}
  token=""
>
  <section>
    <form
      className="form"
      id="loginForm"
      onSubmit={[Function]}
    >
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="loginEmail"
                    label="Your Email"
                    name="email"
                    type="text"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="loginEmail-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="loginEmail"
                      >
                        <span
                          className="Text -primary"
                        >
                          Your Email
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="loginEmail"
                          name="email"
                          onChange={[Function]}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="loginPassword"
                    label="Your Password"
                    name="password"
                    type="password"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="loginPassword-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="loginPassword"
                      >
                        <span
                          className="Text -primary"
                        >
                          Your Password
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="loginPassword"
                          name="password"
                          onChange={[Function]}
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    color="primary"
                    id="forgotPassword"
                    plain={true}
                    size="regular"
                  >
                    <a
                      className="Link -primary -regular -plain"
                      id="forgotPassword"
                    >
                      forgot your password?
                    </a>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    color="primary"
                    fill="solid"
                    id="loginButton"
                    label="Login"
                    size="regular"
                    tabIndex={0}
                    type="submit"
                  >
                    <button
                      className="Button -primary -regular -solid"
                      id="loginButton"
                      tabIndex={0}
                      type="submit"
                    >
                      <div
                        className="Button__inner"
                      >
                        Login
                      </div>
                    </button>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <p
                    id="loginFailure"
                  >
                    Login Failed
                  </p>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
    </form>
  </section>
</Login>
`;
